### PR TITLE
Dev start guide

### PR DESCRIPTION
**Pull request by rodrigovalin**
- [x]  Add kube-config ConfigMap creation to `e2e.py`

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+from a high perspective how everything works together.
+
+After our experience building the [Enterprise MongoDB Kubernetes
+Operator](https://github.com/mongodb/mongodb-enterprise-operator), we have
+realized that is is very important to have a clean environment to work, and as such we have
+adopted a strategy that makes it easier for everyone to contribute.
+
+This strategy is based on using
+[`operator-sdk`](https://github.com/operator-framework/operator-sdk) for running
+the test, and making the test-runner itself to run as a Kubernetes Pod. This
```
Comment (chatton) [inline contributing.md:13]:
```suggestion
the tests, and making the test-runner itself run as a Kubernetes Pod. This
```

File: scripts/dev/e2e.py [hunk 2]
```diff
@@ -72,6 +76,29 @@ def _prepare_testrunner_environment():
...
+    kube_config = yaml.safe_dump(kube_config)
+    data = {"kubeconfig": kube_config}
+    config_map = client.V1ConfigMap(
+        metadata=client.V1ObjectMeta(name="kube-config"), data=data
+    )
+
+    ignore_if_already_exists(
+        lambda: corev1.create_namespaced_config_map("default", config_map)
+    )
+
```
Comment (chatton) [inline scripts/dev/e2e.py:100]: 🙏 Thanks for moving this into the python script! (can we delete it from the bash one, or is that still needed for evg?)

Comment (rodrigovalin) [reply scripts/dev/e2e.py:100]: Still needed for Evergreen runs

Comment (rodrigovalin) [reply scripts/dev/e2e.py:100]: and removed from kind setup!

**Pull request merged**